### ISSUE TITLE
Redefinition of a duplicate syntax binding should be an error

### ISSUE THREAD

**Issue by disnet** [mozilla/sweet.js#539]
Currently a duplicate `syntax` declaration in the same scope will override the previous binding:

```js
syntax m = ctx => #`first`;
syntax m = ctx => #`second`;
m; // second
```

It should be following `let`/`const` binding semantics and throw an early error.

**Comment by gabejohnson** [comment]
Is this the spot?

https://github.com/mozilla/sweet.js/blob/master/src/enforester.js#L1804

**Comment by gabejohnson** [comment]
nm. Didn't see you had self-assigned

**Comment by disnet** [comment]
It's actually here: https://github.com/mozilla/sweet.js/blob/master/src/token-expander.js#L23-L34

But its a bit subtle so I need to think through it.

**Comment by gabejohnson** [comment]
This does throw an error now
```javascript
Error: Scopeset {outsideEdge_1, insideEdge0_2} has ambiguous subsets {outsideEdge_1, insideEdge0_2}, {outsideEdge_1, insideEdge0_2}
```
Though it could probably be a little more helpful.

**Comment by disnet** [comment]
Agreed, in general the ambiguous subsets error needs to be more descriptive (say what and where the binding was) and this case in particular the error should mention re-declaration.
